packages/core/src/credential/Credential.ts

```diff
@@ -201,13 +201,13 @@ export function verifyDataIntegrity(input: ICredential): boolean {
   }
 
   // verify properties against selective disclosure proof
-  const verificationResult = Claim.verifyDisclosedAttributes(input.claim, {
+  const { errors, verified } = Claim.verifyDisclosedAttributes(input.claim, {
     nonces: input.claimNonceMap,
     hashes: input.claimHashes,
   })
   // TODO: how do we want to deal with multiple errors during claim verification?
-  if (!verificationResult.verified)
-    throw verificationResult.errors[0] || new SDKErrors.ClaimUnverifiableError()
+  if (!verified)
+    throw errors.length > 0 ? errors[0] : new SDKErrors.ClaimUnverifiableError()
```

Comment:
There’s no simple way to let TS know that no value exists at some index of array. Empty arrays are also difficult for TS. See [noUncheckedIndexedAccess](https://www.typescriptlang.org/tsconfig#noUncheckedIndexedAccess).